pennylane/ops/op_math/prod.py

```diff
@@ -314,15 +314,19 @@ def eigvals(self):
 
     def matrix(self, wire_order=None):
         """Representation of the operator as a matrix in the computational basis."""
-        wire_order = wire_order or self.wires
         mats = (
-            math.expand_matrix(qml.matrix(op), op.wires, wire_order=wire_order)
+            qml.matrix(op, wire_order=self.wires)
             if isinstance(op, qml.Hamiltonian)
-            else math.expand_matrix(op.matrix(), op.wires, wire_order=wire_order)
+            else op.matrix(wire_order=self.wires)
             for op in self.factors
         )
 
-        return reduce(math.dot, mats)
+        reduced_mat = reduce(math.dot, mats)
+
+        if wire_order is not None:
+            reduced_mat = math.expand_matrix(reduced_mat, self.wires, wire_order=wire_order)
+
+        return reduced_mat
```

Comment:
```suggestion
        return math.expand_matrix(reduced_mat, self.wires, wire_order=wire_order)
```

`math.expand_matrix` now performs a `wire_order is not None` check inside.